modules/standard/GPU.chpl

```diff
@@ -37,6 +37,19 @@ module GPU
   pragma "codegen for CPU and GPU"
   extern proc chpl_gpu_write(const str : c_string) : void;
 
+  pragma "no doc"
+  pragma "codegen for CPU and GPU"
+  extern proc chpl_gpu_clock() : uint;
```

Comment:
Since these functions are meant to be called from device code I need them to be marked with `__device__` and inlined so I put them in `chpl-gpu-gen-includes.h` where we've put other similar functions.

For host functions the there should be a declaration in the vendor-neutral header file and implementations in the vendor-specific source files.

Maybe we should doing something similar for device functions; though I don't think I want to do that for this PR.